### ISSUE TITLE
Single part app pages - Uncaught (in promise) error

### ISSUE THREAD

**Issue by lucmoco** [SharePoint/sp-dev-docs#9322]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
  - Chrome Version 119.0.6045.124 (Official Build) (64-bit)
  - Edge Version 119.0.2151.44 (Official build) (64-bit)
- SPFx version 1.18.1
- Node.js version 16.20.0

### Describe the bug / error

When creating a single part app page showing a custom web part, and checking the console, an error is being logged:
<img width="525" alt="image" src="https://github.com/SharePoint/sp-dev-docs/assets/31615157/ed41d8f1-466d-439b-9366-a014300a45f4">
The error occurs in the javascript file
`https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_681e18935de8c7c2ccb50e213a98d96e.js`,
Also, a warning is logged in the console:
<img width="521" alt="image" src="https://github.com/SharePoint/sp-dev-docs/assets/31615157/f7f8a66d-a946-4ae1-9ecc-291b20cd8f99">
The error and warning are not present if the web part is added to a normal page.

I could reproduce the problem in two different tenants, and using Edge and Chrome.

### Steps to reproduce

1. Create a web part with the Yeoman Generator with the settings:
<img width="314" alt="image" src="https://github.com/SharePoint/sp-dev-docs/assets/31615157/f0c7432c-935e-4ab5-b5e2-7cc7e25ca8a3">

2. Create an SPFx package with the solution (no source code changes required).
3. Create a communication site.
4. Add a site collection app catalog to the site.
5. Add the solution to the app catalog.
6. Create a single part app pages containing the web part.
7. Open the browser's developer tools and see the messages.

### Expected behavior

No error or warning messages are being logged.

**Comment by aminezemzemi** [comment]
Hi, I encounter the same problem.
Did you resolve the problem please ?

**Comment by lucmoco** [comment]
> Hi, I encounter the same problem. Did you resolve the problem please ?

No, more than one year later, I upgraded to SPFx 1.20 and the problem is still there.

**Comment by Ashlesha-MSFT** [comment]
@lucmoco,
We were able to reproduce the issue, and we are investigating it.  We have logged this as a bug, and our engineering team will look into it.

<img width="775" height="550" alt="Image" src="https://github.com/user-attachments/assets/d5b6d6ad-7016-4760-9a2e-a35ee43a73c0" />

Thank you for your patience!

**Comment by VesaJuvonen** [comment]
Hi @lucmoco and @aminezemzemi - we can indeed see the issue in the console and have bypassed the feedback and input to our feedback team, given that the exception comes from their JavaScript implementation. This is nothing really to do with SPFx single app pages, rather the instrumentation of our feedback file is making most likely some assumptions which are not true/correct for single page apps.

As the exception will not block the execution of your SPFx file and is not causing any end user exceptions or challenges, we will be closing the issue from the platform team side - also because we can't really fix this oob feature as dev/platform team does not own that.

We are expecting this to go away hopefully soon - but as it does not cause end user issues, it's more an inconvenience. We appreciate flagging it to clean up the console from unnecessary exceptions and messages.
